diff --git a/Source/_Common/Ctrl.cpp b/Source/_Common/Ctrl.cpp
index d22a34f..f12fa4f 100644
--- a/Source/_Common/Ctrl.cpp
+++ b/Source/_Common/Ctrl.cpp
@@ -211,6 +211,33 @@ BOOL	CCtrl::GetSkillProp( ItemProp **ppSkillProp, AddSkillProp **ppAddSkillProp,
 	return TRUE;
 }
 
+static bool IsMonsterAffectedBySkill(CObj * pObj, CObj * pSrc) {
+
+	if (pObj->GetType() == OT_MOVER) {
+		// 대상이 무버면.
+
+		CMover * pTarget = (CMover *)pObj;
+
+		// NPC가 아닌경우만 적용
+		if (pTarget->IsPeaceful()) return false;
+
+		CMover * pAttacker = (CMover *)pSrc;
+		if (pAttacker->IsPlayer() && !pAttacker->IsChaotic() && pTarget->GetProp()->dwClass == RANK_GUARD)
+			return false;
+
+		return true;
+	}
+
+	if (pObj->GetType() == OT_CTRL) {
+		// 대상이 컨트롤이면.
+		// 
+		// 공격자가 무버일때만 적용. 컨트롤이 컨트롤에게 영향을 줄순 없다.
+		return pSrc->GetType() == OT_MOVER;
+	}
+
+	return false;
+}
+
 
 // vPos를 중심으로 주변에 효과를 적용한다.
 // ------------------------------------------
